src/main/java/com/hms_networks/americas/sc/extensions/historicaldata/RapidCatchUp.java

```diff
@@ -0,0 +1,223 @@
+package com.hms_networks.americas.sc.extensions.historicaldata;
+
+import com.ewon.ewonitf.Exporter;
+import com.hms_networks.americas.sc.extensions.datapoint.DataPoint;
+import com.hms_networks.americas.sc.extensions.json.JSONException;
+import com.hms_networks.americas.sc.extensions.string.StringUtils;
+import com.hms_networks.americas.sc.extensions.system.time.SCTimeUtils;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UnsupportedEncodingException;
+import java.util.List;
+
+/**
+ * This class addresses the need for historical data tracking to "catch up" quickly when there are
+ * periods where the tracker is not running. For example, the Flexy is powered off for several
+ * weeks. For correct behavior, it is required that injectJvmLocalTime from {@link SCTimeUtils} is
+ * called at startup.
+ *
+ * @author HMS Networks, MU Americas Solution Center
+ * @since 1.14.2
+ */
+public class RapidCatchUp {
+
+  /**
+   * Threshold for enabling rapid catch up mode in milliseconds. If current time - start time is
+   * less than this amount, catchup mode will not be enabled.
+   */
+  // 1000 * 60 * 60 = 3,600,000 milliseconds = 1 hour
+  public static final long RAPID_CATCH_UP_TIME_DELTA_MILLISECONDS = 1000 * 60 * 60;
+
+  /** Starting point for request period, which changes based on previous result. */
+  // 1000 * 60 * 10 = 600,0000 milliseconds = 10 minutes
+  public static final long RAPID_CATCH_UP_TIME_BASE = 1000 * 60 * 10;
+
+  /** Maximum amount of time for the request period. */
+  // 144 * 10 minutes = 24 hours
+  public static final long RAPID_CATCH_UP_TIME_DURATION_MAX = 144 * RAPID_CATCH_UP_TIME_BASE;
```

Comment:
Any reason why 1 day was chosen for the max catch up time?

Reply:
Im just wondering if this was chosen arbitrarily or through some testing that should be noted in the comment.

Reply:
The concern is balancing the need to burn through no-data periods quickly with the need to avoid some unknown disaster that results from arriving at data and requesting way too much. I hope that a EBDTimeoutException is the worst that could happen.

I tested 100 tags at 10 seconds and have requested 48 hours of data via HTTP EBD request. This took over 70 seconds but did return. This implementation is quicker, because it just closes the input stream after reading a few chunks, so the request takes around 6 seconds.  So far, I have not seen a EBDTimeoutException. 

So 24 hours is a best guess at a proper balance between burn-through speed, but having a tolerable amount of data when data is found.  Assuming the check is ran at a 10 second interval, if the Flexy is powered off for a year, it will take around an hour to catch up.

Reply:
Would it make sense just adding that text as a comment for this value. I think it adds a lot of value to know why the dev set a value to what they did, especially when its an arbitrary best guess.

Reply:
Yeah, I can add a comment.